### PR TITLE
Replace co-postgres-queries by knex

### LINKED ISSUE #164: Remove co-postgres-queries from the API
This library becomes old, and we need to remove this.

Related to #48

### PR DESCRIPTION
**Pull request by Kmaschta**
Fixes #164 

Tip for the review: disable the whitespace diff, I'm sorry I had to lint a lot of files

## TODO

- [x] Replace projects queries
- [x] Replace environments queries
- [x] Replace configurations queries
- [x] Replace entries queries
- [x] Replace versions queries
- [x] Replace tags queries
- [x] Remove postgres-queries

## Additional

Done!
Also, there is no longer outdated depenency on this project:

image

### CODE REVIEW

File: api/src/queries/configurations.js [hunk 1]
```diff
@@ -1,88 +1,82 @@
...
+      "configuration.name": configurationName,
+      "configuration.state": "live"
     });
-    client.release();
 
-    if (!result.length) {
-        return null;
-    }
-    return result[0];
+  return results[0];
```
Comment (zyhou) [inline api/src/queries/configurations.js]: use `client.first`

File: api/src/queries/configurations.js [hunk 2]
```diff
@@ -1,88 +1,82 @@
...
 };
 
-const findAllByEnvironmentName = async (projectId, environmentName) => {
-    const client = await db.link(query);
-    const result = await client.selectPage(undefined, undefined, {
-        project_id: projectId,
-        'environment.name': environmentName,
-        state: 'live',
+const findAllByEnvironmentName = async (projectId, environmentName) =>
+  client
```
Comment (zyhou) [inline api/src/queries/configurations.js]: add await?

File: api/src/queries/configurations.js [hunk 3]
```diff
@@ -1,88 +1,82 @@
...
-    return configuration;
+  const results = await client
+    .select(fields)
+    .from(table)
+    .innerJoin("environment", "environment.id", "configuration.environment_id")
+    .where({
+      "configuration.id": id
+    });
+
+  return results[0];
```
Comment (zyhou) [inline api/src/queries/configurations.js]: use `client.first`

File: api/src/queries/configurations.js [hunk 4]
```diff
@@ -1,88 +1,82 @@
...
+  const results = await client
+    .select(fields)
+    .from(table)
+    .innerJoin("environment", "environment.id", "configuration.environment_id")
+    .where({
+      "configuration.id": id
+    });
 
-    return configuration;
+  return results[0];
```
Comment (zyhou) [inline api/src/queries/configurations.js]: use `client.first`

File: api/src/queries/environments.js [hunk 5]
```diff
@@ -1,75 +1,36 @@
...
-        project_id: projectId,
-        'environment.name': environmentName,
-        state: 'live',
+  const results = await client
+    .select(fields)
+    .from(table)
+    .where({
+      project_id: projectId,
+      state: "live",
+      name: environmentName
```
Comment (zyhou) [inline api/src/queries/environments.js]: 'environment.name': environmentName, no need to join  ?

File: api/src/queries/projects.js [hunk 6]
```diff
@@ -1,76 +1,33 @@
...
-        parameters: { id, token },
+  const results = await knex
+    .select(fields)
+    .from(table)
+    .where({
+      id,
+      state: "live"
+    })
+    .andWhere(function() {
+      return this.where({ write_token: token }).orWhere({ read_token: token });
```
Comment (zyhou) [inline api/src/queries/projects.js]: this ?

File: api/src/queries/configurations.js (hunk 2 above)
Comment (Kmaschta) [reply api/src/queries/configurations.js]: await is implicit (and lazy) if you return a promise :)

File: api/src/queries/environments.js (hunk 5 above)
Comment (Kmaschta) [reply api/src/queries/environments.js]: Not in my knowledge

File: api/src/queries/projects.js (hunk 6 above)
Comment (Kmaschta) [reply api/src/queries/projects.js]: It seems like the knex way of doing parenthesis:
http://knexjs.org/#Builder-where
(search for Grouped Chain)

**Pull request merged**